### PR TITLE
Added Login, Logout, Sign Up and Follow Capability

### LINKED ISSUE #16: Backend: Create an Account
#11 

Allow the user to create an account through the Sign Up page by entering the appropriate information.

### LINKED ISSUE #17: Backend: Unique Account Verification
#11 

Make sure that users can not make multiple accounts using repeated credentials.

### LINKED ISSUE #18: Backend: Sign In
#11 

Allow a user to sign in to an existing account.

### LINKED ISSUE #21: Back-end: See Other Users - Follow
#11 

From the See Other Users page, have the Follow button set a flag in the backend database.

### LINKED ISSUE #27: Back-end: View Posts of Followed Users on Home Screen
#9 

Let the Home Screen show a list of posts from followed users.

### PR DESCRIPTION
**Pull request by MatteoGisondi**
#11 
Resolves #16, Resolves #17, Resolves #18, Resolves #21, Resolves #27

Functionality put in place for user to submit SignUpForm and LoginForm.
Template provided to allow users to follow other users and view the images associated to their accounts.
Provides a list of users, follower tree and image files in /data.

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -1,13 +1,101 @@
...
+    if form.validate_on_submit():
+        # check first if user already exists
+        user = find_user(form.username.data)
+        if not user:
+            salt = bcrypt.gensalt()
+            password = bcrypt.hashpw(form.password.data.encode(), salt)
+            with open('data/users.csv', 'w') as f:
+                user_data = ','.join(form.username.data, form.email.data, form.first_name.data, form.last_name.data, password.decode())
+                f.write(user_data)
+            flash('Sign up successful.')
```
Comment (LiamBurchellReyes) [inline app.py:93]: Also to be changed to adding to the database (goes along with above TODO on line 82)

File: follower.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+# follow("Drand1943","Milloon")
+# follow("Drand1943","Drand1943")
+# follow("Drand1943","wgerwgg")
+# follow("sdfwergwghr","Drand1943")
+# print(imagesForUser("Giarturner"))
+# print(getUserFollowers("Cagoo1938"))
+# print(getImagesToShow("Whimseeplis"))
+
+
+# *************************************************************************DO NOT USE METHODS***************************************************************************
```
Comment (LiamBurchellReyes) [inline follower.py:150]: Would commenting these methods out be useful at a later point? I see how they can still be useful at this point in the project (Sprint 2), so I am fine with them not being commented out currently.

**Comment by LiamBurchellReyes** [comment]
Requires new issue for password verification. See forms.py - password verification is a tautology currently.

**Review by LiamBurchellReyes — APPROVED** [review]
Upon committing, create a task for Password Validation (assign to me).

File: follower.py (hunk 2 above)
Comment (alexbara2000) [reply follower.py:150]: Yes, once we all discuss the databases I created and no more changes need to be made, these methods will be commented out. For now, we might need to use them so I left them available with a warning to make sure people do not use them without knowing what they are doing

File: app.py [hunk 3]
```diff
@@ -1,13 +1,101 @@
...
+def load_user(user_id):
+    """retrieve a user object for the current user while hiding password"""
+    user = find_user(user_id)
+    if user:
+        user.password = None
+    return user
+
+
+def find_user(username):
+    """ TODO : this will later be changed to searching the database"""
```
Comment (alexbara2000) [inline app.py:37]: You can use the code template in follower.py if you need to find a user in any of the databases.

**Review by alexbara2000 — APPROVED** [review]
All the python files have made good progress and there are no app-breaking issues.

**Pull request merged**